Fix helm issue during e2e test run  operator installation

fjglira opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

    1. If this is your first time, please read our contributor guidelines: https://github.com/istio-ecosystem/sail-operator/blob/main/CONTRIBUTING.md
    2. Discuss your changes before you start working on them. You can open a new issue in the [Sail Operator GitHub repository](https://github.com/istio-ecosystem/sail-operator/issues) or start a discussion in the [Sail Operator Discussion](https://github.com/istio-ecosystem/sail-operator/discussions). By this way, you can get feedback from the community and ensure that your changes are aligned with the project goals.
    3. If the PR is unfinished, make is as a draft.
-->

#### What type of PR is this?
<!--
In order to minimize the time taken to categorize your PR, add a label accoutring to the PR type defined above.
Please, use the following labels, according to the PR type:
    * Enhancement / New Feature - enhancement
    * Bug Fix - bug
    * Refactor - cleanup/refactor
    * Optimization - enhancement
    * Test - test-e2e
    * Documentation Update - documentation
-->

- [ ] Enhancement / New Feature
- [ ] Bug Fix
- [ ] Refactor
- [ ] Optimization
- [x] Test
- [ ] Documentation Update

#### What this PR does / why we need it:
The operator namespace was being created during the image build/push only for OCP test runs, leading to a 'helm name error' in subsequent tests case. This happened because the cleaner record was incorrectly preventing the deletion of the sail-operator namespace.

#### Which issue(s) this PR fixes:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
Add related issue or PR if exists.
-->
Fixes #

Related Issue/PR #

#### Additional information:

mkolesnik commented on tests/e2e/setup/build-and-push-operator.sh:
```diff
@@ -81,3 +81,15 @@ if [ "${OCP}" == "true" ]; then
+
+# Workaround for OCP helm operator installation issues:
+# We create the operator namespace before the tests run
+# To avoid any cleanup issues, after we build annd push the image we check if the namespace exists and delete it if it does.
```
```suggestion
# To avoid any cleanup issues, after we build and push the image we check if the namespace exists and delete it if it does.
```

Small typo

mkolesnik commented on tests/e2e/setup/build-and-push-operator.sh:
```diff
@@ -81,3 +81,15 @@ if [ "${OCP}" == "true" ]; then
+# Workaround for OCP helm operator installation issues:
+# We create the operator namespace before the tests run
+# To avoid any cleanup issues, after we build annd push the image we check if the namespace exists and delete it if it does.
+# The test logic already handle the namespace creation and deletion during the test run. 
```
```suggestion
# The test logic already handles the namespace creation and deletion during the test run. 
```
Small typo

mkolesnik reviewed (commented):
Couple of typos.

Also I'm not sure I understand why this needs to be specifically in the `build-and-push-operator` script and not, for example, the scripts that actually run the tests?
e.g. `integ-suite-ocp.sh`

fjglira commented:
> Couple of typos.
> 
> Also I'm not sure I understand why this needs to be specifically in the `build-and-push-operator` script and not, for example, the scripts that actually run the tests? e.g. `integ-suite-ocp.sh`

Hey, I did some changes:

- Create a new namespace to push the operator image. This will avoid any issues while deleting the operator namespace. Fun fact, this was working before because once we pulled the first time the image from the operator namespace, it was in cache, no matter that we deleted and created again the operator namespace (Started to fail to me today after I started to delete the operator namespace before starting the test)
- Set the proper roles for the new namespace and the operator namespace to be able to pull and push the image
- Moved the deletion of the operator namespace to the common script that actually runs the test after validating that we are running for OCP. Note: we still need to create the operator namespace to create the roles to pull the image

fjglira commented on tests/e2e/setup/build-and-push-operator.sh:
```diff
@@ -37,9 +37,12 @@ get_internal_registry() {
   URL=$(${COMMAND} get route default-route -n openshift-image-registry --template='{{ .spec.host }}')
-  export HUB="${URL}/${NAMESPACE}"
+
+  # Create the istio-images namespace to store the operator image and avoid errors during test run.
```
I'll rephrase this comment

fjglira commented:
/test e2e-kind-multicluster

istio-testing commented:
In response to a cherrypick label: new issue created for failed cherrypick: #935

istio-testing commented:
In response to a cherrypick label: #914 failed to apply on top of branch "release-1.26":
```
Applying: Fix helm issue during e2e test run operator installation
.git/rebase-apply/patch:105: trailing whitespace.
# The test logic already handle the namespace creation and deletion during the test run. 
warning: 1 line adds whitespace errors.
Using index info to reconstruct a base tree...
M	tests/e2e/ambient/ambient_test.go
M	tests/e2e/dualstack/dualstack_test.go
M	tests/e2e/multicluster/multicluster_suite_test.go
M	tests/e2e/multicontrolplane/multi_control_plane_test.go
M	tests/e2e/setup/build-and-push-operator.sh
Falling back to patching base and 3-way merge...
Auto-merging tests/e2e/setup/build-and-push-operator.sh
CONFLICT (content): Merge conflict in tests/e2e/setup/build-and-push-operator.sh
Auto-merging tests/e2e/multicontrolplane/multi_control_plane_test.go
Auto-merging tests/e2e/multicluster/multicluster_suite_test.go
Auto-merging tests/e2e/dualstack/dualstack_test.go
Auto-merging tests/e2e/ambient/ambient_test.go
error: Failed to merge in the changes.
hint: Use 'git am --show-current-patch=diff' to see the failed patch
hint: When you have resolved this problem, run "git am --continue".
hint: If you prefer to skip this patch, run "git am --skip" instead.
hint: To restore the original branch and stop patching, run "git am --abort".
hint: Disable this message with "git config advice.mergeConflict false"
Patch failed at 0001 Fix helm issue during e2e test run operator installation

```

istio-testing commented:
In response to a cherrypick label: new issue created for failed cherrypick: #987
